### PR TITLE
Updated base creative onInit function

### PR DESCRIPTION
**Pull request by ryanthompson591**
This allows users to override onInit and still have access to data changes.

### CODE REVIEW

File: examples/creatives/base_simid_creative.js [hunk 1]
```diff
@@ -20,7 +20,17 @@ class BaseSimidCreative {
...
-    this.videoState = {};
+    this.videoState = {
+      currentSrc:'',
+      currentTime: -1, // Time not yet known
+      duration: -1, // duration unknown
+      ended: false,
+      muted: false,
+      paused: false,
+      volume: 0.5,
+      fullscreen: false //TODO add this to environment data in spec
```
Comment (damargulis) [inline examples/creatives/base_simid_creative.js:31]: I think you can remove the todo now, looks like it is in the spec

File: examples/creatives/base_simid_creative.js [hunk 2]
```diff
@@ -69,21 +79,22 @@ class BaseSimidCreative {
...
+    this.updateInternalOnInit(eventData);
+    this.simidProtocol.resolve(eventData, {});
+  }
+
+  /**
+   * Updates internal data on initialization call.
+   *
+   * Note: When overriding the onInit function and not wishing
+   * to always resolve, subclasses may instead use this function.
+   * @param {!Object} eventData Data from the event.
```
Comment (damargulis) [inline examples/creatives/base_simid_creative.js:91]: `@protected` ?

File: examples/creatives/base_simid_creative.js (hunk 1 above)
Comment (ryanthompson591) [reply examples/creatives/base_simid_creative.js]: Done.

File: examples/creatives/base_simid_creative.js (hunk 2 above)
Comment (ryanthompson591) [reply examples/creatives/base_simid_creative.js:92]: Yeah, I labeled all these functions protected, since they are meant to be overridden.

**Comment by ryanthompson591** [comment]
PTAL.

**Pull request merged**